### ISSUE TITLE
BUG: KubeDB using old unsupported api (need to switch usage of v1alpha1 and v1beta1 to the v1 endpoints)

### ISSUE THREAD

**Issue by willyguggenheim** [willyguggenheim/kubify#3]
# Issue 000001: If namespace deletion is stuck locally in docker desktop's kubernetes:
STATUS: 
This can be RESOLVED with manual click (see image below), but the TECH DEBT automation of this fix is TODO (I/We should automate this fix)
    - TODO (in order to automate the fix): need to find out the cli or api for hard resetting the local docker desktop kubernetes cluster and then add it to ansible and cli
Example error: 
`Error from server (Forbidden): error when creating "STDIN": secrets "dockerhub" is forbidden: unable to create new content in namespace kubify`
Investigation:
- Find out what resources are left in the stuck namespace with: `kubectl api-resources --verbs=list --namespaced -o name | xargs -n 1 kubectl get --show-kind --ignore-not-found -n kubify`
    - Example Output: `error: unable to retrieve the complete list of server APIs: mutators.kubedb.com/v1alpha1: the server is currently unable to handle the request, validators.kubedb.com/v1alpha1: the server is currently unable to handle the request`
        - This was from an old typo in the api version, so I will create a PR to the public https://github.com/kubedb/operator repo to fix this and will try to get a workaorund in place in the meantime #TODO (Willy will take this task)
    - Helpful references of endpoints and versions: 
        - https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/
        - https://kubedb.com/docs/v2021.03.17/guides/postgres/initialization/script_source/
    - RESOLUTION: I hard reset my local Docker Desktop's K8s to fix this (see the picture in DEBUG.md)
![picture from DEBUG.md](https://github.com/willyguggenheim/kubify/raw/master/docs/img/DEBUG_md_imgs/000001/reset_docker_desktop_k8s_stack.png)

^ so 1 TODO for a bug with KubeDB Operator and 1 automation task to avoid manual clicking .. 

Willy will handle both of these items.

**Comment by willyguggenheim** [comment]
@pmtolk fyi: I will fix that and will automate that other technical debt as well ..

**Comment by willyguggenheim** [comment]
I got past the above dependency error, but now I have a blocking error:
`W0407 09:56:52.026785   59725 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition`
I will reach out to the KubeDB slack channel to see if I should create a PR that changes all apiextensions.k8s.io/v1beta1 to apiextensions.k8s.io/v1 or how I should best go about fixing this (or why it's happening) ..

**Comment by willyguggenheim** [comment]
I'll do the find/replace PR of those endpoints from v1betav1 to v1 if I don't hear anything back ........
I am trying Slack as well with a reference to this page, but no one in OS looked at it just yet....

**Comment by willyguggenheim** [comment]
When I try the workaround mentioned in KubeDB's Slack (to ignore the warning for now), I get an actual error:
`error: unable to retrieve the complete list of server APIs: mutators.kubedb.com/v1alpha1: the server is currently unable to handle the request, validators.kubedb.com/v1alpha1: the server is currently unable to handle the request`

**Comment by willyguggenheim** [comment]
The KubeDB public repo latest release is broken heavily:
```
+ helm --kube-context docker-desktop install kubedb-operator appscode/kubedb --version v2021.03.17 --namespace kube-system
W0408 14:10:39.633130   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:39.747216   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:39.867780   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:39.986031   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.106391   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.231380   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.349933   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.469900   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.594823   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.711810   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.876877   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
Error: rendered manifests contain a resource that already exists. Unable to continue with install: PodSecurityPolicy "elasticsearch-db" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-name" must equal "kubedb-operator": current value is "kubedb"
willy.guggenheim@willy-g-macbook-pro kubify % 
```

**Comment by willyguggenheim** [comment]
The KubeDB public repo latest release is broken heavily:

+ helm --kube-context docker-desktop install kubedb-operator appscode/kubedb --version v2021.03.17 --namespace kube-system
W0408 14:10:39.633130   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:39.747216   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:39.867780   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:39.986031   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.106391   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.231380   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.349933   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.469900   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.594823   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.711810   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
W0408 14:10:40.876877   12857 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
Error: rendered manifests contain a resource that already exists. Unable to continue with install: PodSecurityPolicy "elasticsearch-db" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-name" must equal "kubedb-operator": current value is "kubedb"
willy9000@willy-g-macbook-pro kubify %

**Comment by willyguggenheim** [comment]
To clarify this issue. There are 2 issues, but I only want to focus on the blocker:
Issue 1) kubedb blocking Kubernetes teardown on Docker Desktop Kubernetes "reset" button
Issue 2) the latest kubedb production release is broken when using Docker Desktop Kubernetes
I am working on a workaround to issue 2, but I already have a workaround for issue 1

**Comment by willyguggenheim** [comment]
About to push a workaround for this. 
I downgraded KubeDB to the previous version (from March to January) and it started to all work again!! 💯 
Still KubeDB needs to release a fixed version..

**Comment by MRMSTCHS** [comment]
In processEl 20 abr. 2021 7:39 p. m., Willy Guggenheim ***@***.***> escribió:
Closed #3.
